.env.sample

```diff
@@ -1,3 +1,5 @@
 REACT_APP_DROPBOX_CLIENT_ID=your_dropbox_client_id
 REACT_APP_GOOGLE_DRIVE_API_KEY=your_google_drive_api_key
-REACT_APP_GOOGLE_DRIVE_CLIENT_ID=your_google_drive_oauth_client_id
\ No newline at end of file
+REACT_APP_GOOGLE_DRIVE_CLIENT_ID=your_google_drive_oauth_client_id
+REACT_APP_GITLAB_CLIENT_ID=your_gitlab_client_id
+REACT_APP_GITLAB_SECRET=your_gitlab_secret
```

Comment:
Hi Caleb, can you also add the instructions and documentation here in the readme? https://github.com/200ok-ch/organice#setup-any-of-the-synchronization-back-ends
GitLab support should also mentioned in the first sentence of that section.

Reply:
Apologies for not having come up with this question when you initially proposed the UX, but: If I register a OAuth application with Gitlab that is configured for the reference instance https://organice.200ok.ch, am I right in thinking that other users trying to get access to their Gitlab project will be rejected? If so, I propose that we change the login form to include the Gitlab client_id and secret in the form if the variables are not set in the .env file. Like this, if an organisation wants to self-host organice, they can set it once and don't have to share the secret with everyone. At the same time, every Gitlab user can continue to use the reference installation.

What do you think?

Reply:
> am I right in thinking that other users trying to get access to their Gitlab project will be rejected?

@munen No, that's not right (fortunately). Any user of gitlab.com will be able to access their projects. GitLab's "user-owned application" terminology is a little confusing I know, I did a double-take at first too. However that's just referring to who manages the OAuth integration/configuration, not who is able to use it. I tested this with a second GitLab account to verify, too.

So I think we should keep it like this (with you registering an application via `.env`). If someone is interesting in using a self-managed GitLab instance in the future they can either make a PR to add support, or they could deploy it themselves. There's a little more work that would be required to support self-managed instances, such as changing the domain used for API calls, but it wouldn't be too hard if someone is motivated. In the meantime though I would prefer keeping it like this so that people who use gitlab.com don't have to go through the hassle of creating an OAuth application themselves.

---

> Hi Caleb, can you also add the instructions and documentation here in the readme?

@schoettl Oops, true to my developer nature I forgot about that. Done... any suggestions on wording changes are welcome if you see anything that needs improvement.

Reply:
@chasecaleb 

> No, that's not right (fortunately). Any user of gitlab.com will be able to access their projects.

:relieved: 

Then I agree with everything you've said. Thank you for taking the time to explain :pray: